### PR TITLE
STY/TST: include docstring and style linting

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Adds `flake8-docstring` and `hacking` to the automated flake8 linting.

Updates style accordingly throughout the code.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
```
flake8 . --count --select=D,E,F,H,W --show-source --statistics
```

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.2
* `flake8-docstring` and `hacking`

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [na] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
NOTE: I've added H105 (Do not use author tags) to the ignore list.  We haven't talked about what we want in the comments at the top of the files yet, so this could go back in.

File: pysatModels/models/ucar_tiegcm.py [hunk 1]
```diff
@@ -105,8 +105,7 @@ def init(self):
 
 # Required method
 def clean(self):
-    """Method to return UCER/TIE-GCM data cleaned to the specified level, unused
-    """
+    """Return UCER/TIE-GCM data cleaned to the specified level, unused."""
```
Comment (aburrell) [inline pysatModels/models/ucar_tiegcm.py:108]:
```suggestion
    """Return UCAR/TIE-GCM data cleaned to the specified level, unused."""
```

File: setup.py [hunk 2]
```diff
@@ -3,6 +3,12 @@
 # Copyright (C) 2019, Authors
 # Full license can be found in License.md and AUTHORS.md
 # -----------------------------------------------------------------------------
+"""Setup routine for pysat.
+
+Note
+----
+package metadata stored in setup.cfg
+"""
```
Comment (aburrell) [inline setup.py:11]:
```suggestion
package metadata stored in setup.cfg

"""
```

File: setup.py [hunk 3]
```diff
@@ -3,6 +3,13 @@
 # Copyright (C) 2019, Authors
 # Full license can be found in License.md and AUTHORS.md
 # -----------------------------------------------------------------------------
+"""Setup routine for pysat.
+
+Note
+----
+package metadata stored in setup.cfg
```
Comment (aburrell) [inline setup.py:10]:
```suggestion
Package metadata stored in setup.cfg
```

**Review by aburrell — APPROVED** [review]

**Pull request merged**